data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachine.java

```diff
@@ -29,14 +35,15 @@ public class OpensearchStateMachine extends StateMachine<OpensearchState, Opense
     public static final int MAX_REBOOT_FAILURES = 3;
 
     StateMachineTracerAggregator tracerAggregator = new StateMachineTracerAggregator();
+    private static OpensearchProcess process;
 
     public OpensearchStateMachine(OpensearchState initialState, StateMachineConfig<OpensearchState, OpensearchEvent> config) {
         super(initialState, config);
         setTrace(tracerAggregator);
     }
 
-    public static OpensearchStateMachine createNew() {
-
+    public static OpensearchStateMachine createNew(OpensearchProcess process) {
+        OpensearchStateMachine.process = process;
```

Comment:
```suggestion
```

We don't need the process here, in the static variable. I'd suggest removing it.

Reply:
good catch, forgot to remove the static variable which I used in the initial refactoring